### ISSUE TITLE
Graceful shutdown

### ISSUE THREAD

**Issue by vdakalov** [Lightstreamer/Lightstreamer-lib-client-javascript#1]
Hello. Tell me please, how I can clear this timer?

https://github.com/Lightstreamer/Lightstreamer-lib-client-javascript/blob/6e549e4616f92a3924ec64114d65d792dbb12e99/src-tool/Executor.js#L104

It's need for clear task queue in event loop and graceful shutdown. I studied your code, but I not found public method to make it.

**Comment by ghost** [comment]
Hello,

Executor.js is not a public module and it is not meant to be a general purpose scheduler, so its API only exposes the operations that are strictly needed by the other modules. 
On the other hand, if you need to be able to shutdown the executor, it is not difficult to make such an extension.

**Comment by vdakalov** [comment]
Can you show me something about extensions: examples, documentation, etc. I don't understand what is `extension` for lightstreamer. Thanks.

**Comment by weswit-team** [comment]
I mean that the code is public, so you can modify it however you want.

**Comment by vdakalov** [comment]
Oh. Okay. What do you think, this (https://github.com/vdakalov/Lightstreamer-lib-client-javascript/commit/0f670d7a81c312a81dd12f48c19f0cc37973d107) changes is safe?

**Comment by ghost** [comment]
Hello Viktor, 

Maybe I have misunderstood your requirements.
Do you want to destroy the Excutor because (1) you have finished with the LightstreamerClient and you want to release all the resources, or (2) you have found that the Executor survives to the closing of the page and you fear there is some kind of leakage?

**Comment by vdakalov** [comment]
No, I'm just want to allow nodejs process made graceful shutdown. This is not possible because `Executor` module has active interval. So, I have to make hard stopping with `process.exit()`. Otherwise process continue working and this is bad. I'm not work with the library in browser, Node.js only. Also leaks doesn't bother me.

**Comment by vdakalov** [comment]
This commit solve my problem https://github.com/vdakalov/Lightstreamer-lib-client-javascript/commit/f8a8e0fd7d0546387745e68674ac06d684e64184

**Comment by ghost** [comment]
Hello Viktor,

That makes sense. We will try to add the feature in a next release of the client.

**Comment by vdakalov** [comment]
Thank you. I thing it's a single way. Because when I build my own client with other version  and build I have error `71 License not valid for this Client type`. When I set original `LS_CID` value, client start works. I understand correctly, any time when you release a new version, it will broken other clients with own LS_CID hash?

**Comment by ghost** [comment]
Indeed, the rule is that a custom client cannot have a version greater than the last released. Otherwise the server will not accept it. For example, since the last released version is 8.0.2, you cannot use 8.1.0. Our advice is to choose a version having 0 as major number (e.g. 0.X.Y).

We will make the build instructions clearer about this point.
